app/src/main/java/org/simple/clinic/drugs/EditMedicinesEffectHandler.kt

```diff
@@ -42,9 +42,13 @@ class EditMedicinesEffectHandler @AssistedInject constructor(
 
       val protocolDrugsStream = effects
           .observeOn(io)
-          .flatMap { userSession.requireLoggedInUser() }
-          .switchMap { facilityRepository.currentFacility(it) }
+          .map {
+            val loggedInUser = userSession.loggedInUserImmediate()!!
+            facilityRepository.currentFacilityImmediate(loggedInUser)!!
+          }
           .switchMap { protocolRepository.drugsForProtocolOrDefault(it.protocolUuid) }
+          .observeOn(schedulersProvider.ui())
```

Comment:
Is there a reason this is being observed on the UI thread?

Reply:
Where does the thread get switched? I wanted to ask this before as well.
Are the events already emitted on the `mainThread`?

Reply:
Since we have a `.observeOn(schedulersProvider.ui())` there, any event that gets emitted will be emitted on the UI thread.

I just wanted to know why this line was added.